core/src/main/scala/kafka/admin/LogDirsCommand.scala

```diff
@@ -39,19 +39,31 @@ object LogDirsCommand {
     def describe(args: Array[String], out: PrintStream): Unit = {
         val opts = new LogDirsCommandOptions(args)
         val adminClient = createAdminClient(opts)
-        val topicList = opts.options.valueOf(opts.topicListOpt).split(",").filter(!_.isEmpty)
-        val brokerList = Option(opts.options.valueOf(opts.brokerListOpt)) match {
-            case Some(brokerListStr) => brokerListStr.split(',').filter(!_.isEmpty).map(_.toInt)
-            case None => adminClient.describeCluster().nodes().get().asScala.map(_.id()).toArray
-        }
+        val topicList = opts.options.valueOf(opts.topicListOpt).split(",").filter(_.nonEmpty)
+        var nonExistBrokers: Set[Int] = Set.empty
+        try {
+            val clusterBrokers: Set[Int] = adminClient.describeCluster().nodes().get().asScala.map(_.id()).toSet
+            val brokerList = Option(opts.options.valueOf(opts.brokerListOpt)) match {
+                case Some(brokerListStr) =>
+                    val inputBrokers: Set[Int] = brokerListStr.split(',').filter(_.nonEmpty).map(_.toInt).toSet
+                    nonExistBrokers = inputBrokers.diff(clusterBrokers)
+                    inputBrokers
+                case None => clusterBrokers
+            }
```

Comment:
nit: We usually avoid using mutable variable unless it is really necessary. In this case, I would rather return the `nonExistingBrokers` when the argument is processed. Something like this:

```
val (existingBrokers, nonExistingBrokers) = Option(opts.options.valueOf(opts.brokerListOpt)) match {
    case Some(brokerListStr) =>
        val inputBrokers: Set[Int] = brokerListStr.split(',').filter(_.nonEmpty).map(_.toInt).toSet
       (inputBrokers, inputBrokers.diff(clusterBrokers)
    case None => (clusterBrokers, Set.empty)
}
```